src/app/components/movie-details/movie-details.component.ts

```diff
@@ -5,18 +5,45 @@ import {
 } from '@angular/common';
 import {
   Component,
-  Input
+  Input,
 } from '@angular/core';
-import { MovieDetails } from '../../shared/interfaces/movies';
+import { take } from 'rxjs';
+import { ApiRequestType } from '../../shared/enums/api-request';
+import { MediaType } from '../../shared/enums/media-types';
+import {  MovieDetails } from '../../shared/interfaces/movies';
 import { ImagePathPipe } from '../../shared/pipes/image-path.pipe';
+import { ApiService } from '../../shared/services/api.service';
+import { GuestSessionService } from '../../shared/services/guest-session.service';
+import { MovieRatingComponent } from '../movie-rating/movie-rating.component';
 
 @Component({
   selector: 'tmbd-movie-details',
   standalone: true,
-  imports: [ImagePathPipe, NgIf, NgForOf, DatePipe],
+  imports: [ImagePathPipe, NgIf, NgForOf, DatePipe, MovieRatingComponent],
   templateUrl: './movie-details.component.html',
   styleUrls: ['./movie-details.component.scss']
 })
 export class MovieDetailsComponent {
   @Input() public details?: MovieDetails;
+  @Input() public pageType: MediaType.Movie | MediaType.Tv = MediaType.Movie;
+
+  private guestSession = this.guestSessionService.guestSession;
+  public isVoted = false;
+
+  constructor(
+    private readonly apiService: ApiService,
+    private readonly guestSessionService: GuestSessionService,
+  ) {}
+
+  public rateMovieOrTv(rating: number): void {
+    const rateParams = `${this.pageType}/${this.details?.id}/${ApiRequestType.Rating + this.guestSession}`;
+
+    this.apiService.rateMovieOrTv$(rateParams, {
+      value: rating,
+    })
+    .pipe(
+      take(1),
```

Comment:
If you leave the page, then `take(1)` will not safe you and the request still be alive. Please, you ngOnDestroy to unsubscribe.